### PR TITLE
Replace guzzle3 with egeloen/ivory-http-adapter

### PR DESCRIPTION
**Pull request by tyx**

### CODE REVIEW

File: src/Rest/RestApiBrowser.php [hunk 1]
```diff
@@ -70,62 +110,48 @@ public function sendRequest($method, $url, $body = null, array $options = array(
...
-            }
-            $this->requestHeaders[$name][] = $value;
-        } else {
-            $this->requestHeaders[$name] = $value;
+        if (!$this->hasHost($uri)) {
+            $uri = rtrim($this->httpClient->getConfiguration()->getBaseUri(), '/') . '/' . ltrim($uri, '/');
         }
+        $stream = new Stream('php://memory', 'rw');
+        if ($body) {
+            $stream->write($body);
```
Comment (shouze) [inline src/Rest/RestApiBrowser.php]: @tyx, @dsamblas   here body must absolutely be a string in fact... as we use [Zend\Diactoros\Stream::write()](https://github.com/zendframework/zend-diactoros/blob/c3b31a5040bef3f73f0348df607f2a0bd978723a/src/Stream.php#L204) method specified by [StreamInterface:write()](https://github.com/php-fig/http-message/blob/85d63699f0dbedb190bbd4b0d2b9dc707ea4c298/src/StreamInterface.php#L115).

So... if we accept resources & arrays, we have to:
- Convert them into string.
- Add corresponding unit tests for that.

File: src/Rest/RestApiBrowser.php [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
 namespace Rezzza\RestApiBehatExtension\Rest;
 
-use Guzzle\Http\Exception\BadResponseException;
-use Guzzle\Http\ClientInterface as HttpClient;
-use Guzzle\Http\Message\Response;
+use Ivory\HttpAdapter\HttpAdapterFactory;
+use Ivory\HttpAdapter\HttpAdapterInterface as HttpClient;
+use Ivory\HttpAdapter\HttpAdapterException;
+use Ivory\HttpAdapter\Message\Request;
+use Zend\Diactoros\Stream;
```
Comment (shouze) [inline src/Rest/RestApiBrowser.php]: should be instead:
```php
use Psr\Http\Message\ StreamInterface;
```

Comment (tyx) [inline src/Rest/RestApiBrowser.php]: We need to instantiate a real implementation in fact. So we need the `Zend\Diactoros\Stream`

File: src/Rest/RestApiBrowser.php (hunk 1 above)
Comment (tyx) [inline src/Rest/RestApiBrowser.php]: I don't see any good reason for now to support ressource or array. Best option is to change phpdoc for now I guess.

Comment (shouze) [inline src/Rest/RestApiBrowser.php]: In fact in one of our apps, if we use guzzle or curl adapter we receive here an `array`. I need to investigate why.

**Pull request merged**